addons/binding/org.openhab.binding.blebox/src/main/java/org/openhab/binding/blebox/devices/BaseDevice.java

```diff
@@ -0,0 +1,133 @@
+
+            if (requestObject.getRootElement() != null) {
+                jo.add(requestObject.getRootElement(), je);
+
+                json = jo.toString();
+            } else {
+                json = je.toString();
+            }
+
+            final Request request = Request.Post(url).connectTimeout(2000).socketTimeout(2000).bodyString(json,
+                    ContentType.APPLICATION_FORM_URLENCODED);
+
+            Content response = request.execute().returnContent();
+
+            if (responseRoot != null) {
+                JsonParser p = new JsonParser();
+                JsonElement jsonContainer = p.parse(response.asString());
+                JsonElement jsonQuery = ((JsonObject) jsonContainer).get(responseRoot);
+
+                return gson.fromJson(jsonQuery, type);
+            } else {
+                return gson.fromJson(response.asString(), type);
+            }
+        } catch (Exception ex) {
+            return null;
+        }
+    }
+
+    public <T extends BaseResponse> T getJson(String path, Class<T> type, String responseRoot) {
+        try {
+            Gson gson = new Gson();
+            URI url = buildUrl(path);
+
+            final Request request = Request.Get(url).connectTimeout(2000).socketTimeout(2000);
+            Content response = request.execute().returnContent();
+
+            if (responseRoot != null) {
+                JsonParser p = new JsonParser();
+                JsonElement jsonContainer = p.parse(response.asString());
+                JsonElement jsonQuery = ((JsonObject) jsonContainer).get(responseRoot);
+
+                return gson.fromJson(jsonQuery, type);
+            } else {
+                return gson.fromJson(response.asString(), type);
+            }
+        } catch (Exception ex) {
+            return null;
+        }
+    }
+
+    public <T extends BaseResponse> T[] getJsonArray(String path, Class<T[]> type, String responseRoot) {
+        try {
+            Gson gson = new Gson();
+            URI url = buildUrl(path);
+
+            final Request request = Request.Get(url).connectTimeout(2000).socketTimeout(2000);
+            Content response = request.execute().returnContent();
+
+            if (responseRoot != null) {
+                JsonParser p = new JsonParser();
+                JsonElement jsonContainer = p.parse(response.asString());
+                JsonElement jsonQuery = ((JsonObject) jsonContainer).get(responseRoot);
+
+                return gson.fromJson(jsonQuery, type);
+            } else {
+                return gson.fromJson(response.asString(), type);
+            }
+        } catch (Exception ex) {
+            return null;
+        }
+    }
+
+    public URI buildUrl(String path) {
+        URIBuilder builder = new URIBuilder();
+        builder.setScheme("http").setHost(IpAddress).setPath(path);
+        URI requestURL = null;
+        try {
+            requestURL = builder.build();
+        } catch (URISyntaxException use) {
+        }
```

Comment:
Empty catch blocks hide problems. If happening you won't know why, so better log to warn or let method throw exception.